packages/repository-tests/src/crud/relations/acceptance/has-many.relation.acceptance.ts

```diff
@@ -0,0 +1,253 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect, toJSON} from '@loopback/testlab';
+import * as _ from 'lodash';
+import {
+  CrudFeatures,
+  CrudRepositoryCtor,
+  CrudTestContext,
+  DataSourceOptions,
+} from '../../..';
+import {
+  deleteAllModelsInDefaultDataSource,
+  withCrudCtx,
+} from '../../../helpers.repository-tests';
+import {Customer, Order} from '../fixtures/models';
+import {givenBoundCrudRepositories, mixedIdType} from '../helpers';
+
+export function hasManyRelationAcceptance(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: CrudRepositoryCtor,
+  features: CrudFeatures,
+) {
+  describe('HasMany relation (acceptance)', () => {
```

Comment:
I am little bit confused about why we have two test files for `HasMany` relation?
- `describe('HasMany relation without DI (acceptance)',`
- `describe('HasMany relation (acceptance)'`

It's not clear to me what is the difference.

I feel `repository-tests` should not be dealing with different ways how relation repositories can be wired up (e.g. with DI or without DI). `repository-tests` should be focused on test cases verifying behavior that can be database specific and thus we want to test it against different databases. Whether the repositories are wired together using DI or not, that's something to test elsewhere - probably back in `repository`. 

Having said that, I am not sure if the tests in this file are still useful - maybe we can simply delete this file?

Reply:
You're right. Before we changing the repositories to functions, those repos were using injection. We removed the injection part so that the repo classes can be created by functions `create{Name}RepoClass`. 
Will move this file back to `repository`.